bundles/org.openhab.binding.amberelectric/src/main/java/org/openhab/binding/amberelectric/internal/AmberElectricWebTargets.java

```diff
@@ -0,0 +1,86 @@
+/**
+ * Copyright (c) 2010-2024 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.amberelectric.internal;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Properties;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.amberelectric.internal.api.CurrentPrices;
+import org.openhab.binding.amberelectric.internal.api.Sites;
+import org.openhab.core.io.net.http.HttpUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Handles performing the actual HTTP requests for communicating with the AmberAPI.
+ *
+ * @author Paul Smedley - Initial Contribution
+ *
+ */
+@NonNullByDefault
+public class AmberElectricWebTargets {
+    private static final int TIMEOUT_MS = 30000;
+
+    private final Logger logger = LoggerFactory.getLogger(AmberElectricWebTargets.class);
+
+    public AmberElectricWebTargets() {
+    }
+
+    public Sites getSites(String apikey, String nmi) throws AmberElectricCommunicationException {
+        String getSitesUri = "https://api.amber.com.au/v1/sites";
+        String response = invoke("GET", getSitesUri, apikey);
+        return Sites.parse(response, nmi);
+    }
+
+    public CurrentPrices getCurrentPrices(String siteid, String apikey) throws AmberElectricCommunicationException {
+        String getCurrentPricesUri = "https://api.amber.com.au/v1/sites/" + siteid + "/prices/current";
```

Comment:
I would suggest to extract "https://api.amber.com.au/v1/sites" to a constant, since the same is also used in the method above.

Reply:
Done.